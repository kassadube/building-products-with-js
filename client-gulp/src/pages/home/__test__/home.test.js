/* eslint-disable react/react-in-jsx-scope */

import configureMockStore from 'redux-mock-store';
import HomeWarraper, {Home} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: '2017-05-07T10:22:37.364Z',
};

const question = {
  id: '0',
  text: 'test',
  owner: user,
  expirationDate: new Date(2016, 1, 1, 1, 1, 1, 1), //'2017-05-07T10:22:37.364Z',
};


test('# Home Warraper', () => {
  const store = mockStore({auth: user, questions: {questions: []}});
  const component = (<HomeWarraper user={user} store={store} />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});


test('# Home', () => {
  const component = (<Home />);
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});

test('# Home with Questions', () => {
  const questions = [];
  //const questionsnew = [];
  const fetchQuestions = () => {
    questions.push({ ...question, id: 5});
  };

  const component = (
    <Home user={user} fetchQuestions={fetchQuestions} questions={questions} addAnswer={() => {}} />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});

