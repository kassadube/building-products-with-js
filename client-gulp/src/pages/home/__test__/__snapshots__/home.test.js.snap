// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Home 1`] = `
<div>
  <Component
    user={Object {}}
  />
  <div />
</div>
`;

exports[`# Home Warraper 1`] = `
<Home
  addAnswer={[Function]}
  fetchQuestions={[Function]}
  questions={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={Object {}}
/>
`;

exports[`# Home with Questions 1`] = `
<div>
  <Component
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": "2017-05-07T10:22:37.364Z",
      }
    }
  />
  <div>
    <Question
      onAnswer={[Function]}
      question={
        Object {
          "expirationDate": 2016-01-31T23:01:01.001Z,
          "id": 5,
          "owner": Object {
            "id": "0",
            "login": "test",
            "registrationDate": "2017-05-07T10:22:37.364Z",
          },
          "text": "test",
        }
      }
    />
  </div>
</div>
`;
